adr/0250-developing-and-publishing-clientside-behaviours.md

```diff
@@ -0,0 +1,48 @@
+# 0250. Developing and Publishing ClientSide Behaviours
```

Comment:
👋🏻 one other thought here: have we considered if we might want to provide some sort of encapsulation of the Javascript we write for ViewComponents? I could see us coupling the two together in similar fashion to our explorations of CSS modules, generating a selector specifically for a component and using it in the JS.

Reply:
Our current architecture is leveraging Web Components, which are encapsulated by tag name. Attempting to define an already defined web component causes an exception on the front-end. 

Web Components are encapsulated in that they are only activated for/on the pre-defined tags. In other words behaviour is not executed globally, only when an element enters the dom.

Not in the scope of this ADR but something I want to explore further is adding a suite of meta-tests which assert that the JS follows good conventions, such as:

 - The naming of JS files matches the naming of components.
 - The JS avoids global behaviours where applicable.
 - Any component with a JS file must have an integration test.

Reply:
> Attempting to define an already defined web component causes an exception on the front-end.

This is the aspect I'm curious about. Can we provide an architecture where this exception is raised at compile time or as part of a test suite?

(This isn't a blocker at all)

Reply:
This is already implicitly part of our test suite as we include the compiled bundle in the system tests!